### PR TITLE
Feature: BB-244 move notification failed processing logic to replication status processor

### PR DESCRIPTION
**Pull request by welansari**
Issue: [BB-244](https://scality.atlassian.net/browse/BB-244)

This refactoring is needed for multiple reasons:
- Send “replication failed” notification only after CRR replay has failed
- Avoid duplicating “resources” (like NotificationConfigManager) in each instance of QueueProcessor
- Keep all status processor in a single place

### CODE REVIEW

**Review by KillianG — APPROVED** [review]

File: extensions/replication/tasks/UpdateReplicationStatus.js [hunk 1]
```diff
@@ -102,6 +115,9 @@ class UpdateReplicationStatus extends BackbeatTask {
             score,
         };
         this.failedCRRProducer.publishFailedCRREntry(JSON.stringify(message));
+        if (this.bucketNotificationConfig) {
```
Comment (francoisferrand) [inline extensions/replication/tasks/UpdateReplicationStatus.js]: this function may not be called, e.g. if `repConfig.monitorReplicationFailures` is not set : but we should still post a bucket notification in that case.

File: tests/unit/replication/ReplicateObject.spec.js [hunk 2]
```diff
@@ -1,166 +0,0 @@
-const assert = require('assert');
```
Comment (francoisferrand) [inline tests/unit/replication/ReplicateObject.spec.js:1]: we should port these tests to some `tests/unit/replication/UpdateReplicationStatus.js` ?

**Review by ghost — APPROVED** [review]
LGTM! Can we mention in the ticket that the `tests` workflow was updated with GHA caching? Just to keep track of why it was done in this PR

**Review by francoisferrand — APPROVED** [review]

**Pull request merged**